src/otx/core/model/base.py

```diff
@@ -164,7 +164,7 @@ def training_step(self, batch: T_OTXBatchDataEntity, batch_idx: int) -> Tensor:
                     prog_bar=True,
                 )
 
-            total_train_loss = sum(train_loss.values())
+            total_train_loss = train_loss.get("loss", sum(train_loss.values()))
```

Comment:
So, it's an introduction of new way to set final loss, right? Doesn't it affect other models? because "loss" looks too general word, I think some models which have set "loss" in dict can be affected.

Reply:
I'll check it and leave here.

Reply:
I tried to find what will be affected by this change and I found visual prompting task only.
But vpm uses `loss` as total loss that is computed with specific way (`20 * focal loss + dice loss + iou loss`), not just using summation.

But I think this change looks more general because `sum(train_loss.values())` doesn't consider whether `train_loss` has `loss` key or not, assuming `loss` is used as total loss.
For example, it seems that vpm loss has been calculated incorrectly due to this part.
Since https://github.com/openvinotoolkit/training_extensions/pull/3155, vpm total loss has been computed twice.
For example, if focal loss=1, dice loss=1, and iou loss=1, total loss should be 20 * focal loss + dice loss + iou loss = 22.
But current total loss is computed once more in that part, so it becomes 22 + 1 + 1 + 1 = 45.
Due to this change, it will return to normal.

Reply:
so it means that the only model affected is VPM and it has miscalculated loss and it can be solved by current change, right?

Reply:
It means that we cannot use just "loss" in our models, since only this "loss" can be optimized at the end. Can't we avoid this behavior or at least rename it like "summary_loss", "sum_loss", etc.

Reply:
I agree with your suggestion, @kprokofi, it seems the easiest way for the final loss decision.
But I have two concerns:
1. It may make the logging loss part more complicated. Referring to the below code, if `train_loss` has `loss` key, its value is logged at L159 with `train/loss` and it is a correct value. But at L168, a new loss with all values added is overwritten to `train/loss`. We can resolve this issue by changing the key name at L168 as you said (assuming it to `sum_loss`), but if there is a task that uses `loss` as a single loss, we have two keys just for this single loss, `loss` and `sum_loss`.
https://github.com/openvinotoolkit/training_extensions/blob/709d841f4a2c9e8dc73b5e416a1deb7536a6b761/src/otx/core/model/base.py#L157-L175

2. Actually, I'm not sure that `training_step` is a proper location for loss summation. I think that the whole process of loss calculation must be done in loss modules. I'd like to suggest removing L167~174, unifying the key of final loss to `loss`, and just returning the value of `loss` as the final loss. For this, every loss module should be updated and it is not within the scope of this PR.

Reply:
> so it means that the only model affected is VPM and it has miscalculated loss and it can be solved by current change, right?

Correct.

Reply:
I agree with the opinion that final loss should be calculated in loss module. I think current problem comes from that same way is used for logging a loss and calculating a loss. If we separate it and change final loss to be calculated in loss module, it can be better.

Reply:
I think deciding how to deal with it is out of this PR, but I agree with Kirill's opinion. Developers have to know setting "loss" mean that that value will be used as final loss. I think it is likely to be missed. To avoid it, how about changing it to "sum_loss" or others values that is difficult to set by accident for now?

Reply:
I agree with your opinions about distinguishing the final loss with others.
I'd like to suggest `total_loss`, more general name convention than `sum_loss`.
Because some tasks use weighted summation loss, not just adding losses like this YOLOv9.

And the whole process of finding the loss must be done within the loss modules as I said.
Let's do it in the next PR.